### ISSUE TITLE
[검수] Ch 21 — 작은 BERT 분류 — 영어 Yelp 이진 (일반 도메인 사전학습 → 다른 도메인 fine-tune)

### ISSUE THREAD

**Issue by yoon-gu** [yoon-gu/neuqes-101#45]
## 담당과 일정

- 기본 페어: C
- 세 번째 검수자: 해당 없음
- Phase: Phase 3
- Required reviews: 2
- Target date: 2026-08-16

## 기준 버전

- Source notebook: `21_en_bert_classify/21_en_bert_classify.ipynb`
- Source commit:
- WikiDocs URL:
- EPUB build:

## Notebook

- [ ] 1차 검수자 PASS
- [ ] 2차 검수자 PASS

## WikiDocs

- [ ] 1차 검수자 PASS
- [ ] 2차 검수자 PASS

## EPUB

- [ ] 1차 검수자 PASS
- [ ] 2차 검수자 PASS

## 완료 조건

- [ ] Blocker 0건
- [ ] Major 0건
- [ ] 모든 수정사항 재검수 완료
- [ ] 최종 Source commit과 EPUB build 기록

상세 운영 기준: #24

**Comment by yum-yeom** [comment]
WikiDocs 1차 PASS (C1)
검수자: @yum-yeom (염민혁, C1)
기준 커밋: 61a71df (PR #144 squash 머지) / 사이트 확인: 2026-08-27
확인: 린트 E1~E9 160파일 0건 · 21-1~21-4 목차↔페이지 대응 · 본편·부록 수치 = executed 실행본 대조 일치 · 21-4 부록 신설(코드 16셀·출력 15·이미지 2 일치) · 부록 링크 4곳 Colab URL 전환 · IPython HTML 객체 잔존 0 · 깨진 문자 0
발견 이슈: 부록 산문 모순 1건 → PR #144 에 정정 포함 (afc8632, #128 후속)

**Comment by yum-yeom** [comment]
Notebook 1차 PASS (C1)
검수자: @yum-yeom (염민혁, C1)
기준 커밋: 61a71df
환경: Google Colab T4 (colab-cli, PyTorch 2.11.0+cu128 / Tesla T4)
실행 시간: 본편 2분 8초 + 부록 2분 47초 (30분 제약 내)
실측: 본편 accuracy 0.6320 · MLM eval loss 7.2212 / perplexity 1368.08 (커밋 실행본 0.6310 / 7.2214 / 1368.41 — 시드 검증 폭 내) · 부록 A 0.6230 / B 0.7820 / C 0.5960 — A−C +0.0270 순 효과 실재, B 우위 결론 유지 (B·C는 커밋 실행본과 소수 4자리까지 동일)
확인: 본편·부록 마지막 셀(비교 차트·§6 표)까지 도달 · 에러 0 · 출력 육안 확인(깨진 문자·이상 반복 없음) · bf16/flash_attention 없음, fp16 사용 · 본문 서술 = 셀 출력 대조
발견 이슈: #121 #126 #127 #128 #129 (+ #64·#123·#143 해당) — 수정 반영·재검수 완료

**Comment by fluentmin** [comment]
**Notebook 2차 PASS (C2)**
검수자: @fluentmin (유창민, C2)
기준 커밋: `4863c9e` (Ch 21 노트북·부록은 `61a71df` 와 동일 — 이후 커밋은 wikidocs 변환기·페이지만 수정)
환경: Google Colab T4 (colab-cli, PyTorch 2.11.0+cu128 / Tesla T4, driver 580.82.07 · CUDA 13.0)
실행 시간: 본편 2분 32초 + 부록 2분 54초 (30분 제약 내)

**실측 (괄호 = 커밋 실행본)**
- 본편: accuracy **0.6320** (0.6310) · precision 0.6066 (0.6055) · recall 0.6818 (동일) · F1 0.6420 (0.6414) · AUC 0.6806 (0.6808)
- 본편 MLM: eval loss 7.2211 (7.2214) / perplexity 1368.05 (1368.41)
- 부록: A **0.6240** (0.6250) · B **0.7820** (동일, F1·AUC 도 소수 4자리까지 동일) · C **0.5960** (동일)
- 부록 delta: A−C **+0.0280** (커밋 +0.0290) — 사전학습 순 효과 실재, A−B −0.1580 으로 **B 우위 결론 유지**
- @yum-yeom 님 실측(본편 0.6320)과도 일치 — 흔들리는 폭이 소수 3자리 이하로 시드 검증 폭 내

**확인**
- 본편 코드셀 24개 전부 실행(출력 22 · 이미지 3), 부록 16개 전부 실행(출력 15 · 이미지 2) — 커밋 실행본과 셀·출력·이미지 개수 일치
- error 출력 0 · 미실행 셀 0 · 마지막 셀(본편 §6 Ch 10 비교표, 부록 accuracy delta)까지 도달
- `fp16=True` 확인, `bf16`·`flash_attention` 사용 없음
- 본편 실행본의 `source_sha256` 도장이 이번 실행과 동일(`172671a…`) — 같은 소스에서 나온 출력임을 확인
- 부록은 도장이 8/20 실행 시점 기준이라 값은 다르지만, 소스 노트북과 코드·마크다운 셀 내용이 완전히 동일(#128 후속 마크다운 동기화가 도장 이후에 이뤄진 것) — 실행 대상은 같습니다

**발견 이슈**: 실행이 아니라 변환 쪽 1건 — WikiDocs 에서 §「숫자로 감 잡기」표가 인용(`>`) 블록 안에 있어 표로 파싱되지 않고 파이프까지 한 문단으로 뭉개졌습니다. 변환기(`build_wikidocs.py`)가 인용 속 표를 인용 밖으로 꺼내도록 수정하고 Ch 21 페이지에 반영했습니다 (`4863c9e`).

**Comment by fluentmin** [comment]
WikiDocs 2차 PASS (C2)
검수자: @fluentmin (유창민, C2)
기준 커밋: `4863c9e` / 사이트 확인: 2026-08-29

검수 도중 발견 이슈: 인용 블록 안 표 파싱 문제(위 `4863c9e` 커밋을 통해 변환기 및 페이지 수정 후 사이트 확인 완료)

이슈 수정 전후 위키독스 내 해당 부분 이미지

(수정 전)
<img width="697" height="71" alt="Image" src="https://github.com/user-attachments/assets/1dcee47f-dea4-4969-9d9f-a17d8e618c54" />

(수정 후)

<img width="453" height="276" alt="Image" src="https://github.com/user-attachments/assets/eac6c8fc-0698-4e1d-bfb2-eaed7a62e10c" />

**Comment by fluentmin** [comment]
@yum-yeom 노트북과 위키독스 확인 감사드립니다. 위키독스에서 표 파싱이 안 되는 문제가 있어 해당 부분만 위 코멘트와 같이 수정하였습니다. 
EPUB 관련 변환기 문제 해결되고 난 후 tex 확인하고 이 이슈 닫으시지요!

**Comment by yum-yeom** [comment]
**EPUB 1차 PASS (C1)**
- 검수자: @yum-yeom (염민혁, C1) / 기준 커밋: `30a5cae` / 빌드: pandoc 3.10.2, `./book/build-epub.sh` (macOS)
- 확인: Apple Books 리더 — 글자 깨짐·표/그림·목차 이동 이상 없음. 자동 점검: 34장 목차 앵커 완비, 가시 텍스트 raw 수식 매크로 0건 (#145 해소 확인)
- 발견 이슈: 없음

@fluentmin EPUB 2차 PASS 기록까지 남겨주시면 그때 닫겠습니다 (산출물별 2인 PASS 충족).

**Comment by fluentmin** [comment]
**EPUB 2차 PASS (C2)**

- 검수자: @fluentmin (유창민, C2) / 기준 커밋: `4a3ae77` / 빌드: pandoc 3.9.0.2, `./book/build-epub.sh` (macOS)
- **기준 커밋 참고**: @yum-yeom 님 1차 PASS 기준(`30a5cae`) 이후 Ch 21 관련 파일(`21_en_bert_classify/*`, `book/chapters/ch21*`, `book/assets/figures/ch21*`, `pages/21-*`, `executed/21_*`·`appendix_compute_budget.ipynb`)은 **변경 0건**입니다. 그 사이 `book/` 변경은 `ch23_ko_bert_classify.tex` 하나뿐이라, 1차와 2차가 같은 Ch 21 내용을 본 것이 맞습니다.
- **빌드**: exit 0, 경고·에러 0줄. preview / publisher-draft / compact 3종 정상 생성.
- **확인 (Apple Books 리더)**: 글자 깨짐·표·그림·목차 이동 이상 없음.

Notebook·WikiDocs·EPUB 모두 2인 PASS 충족입니다. @yum-yeom 확인 후 닫아 주세요.

**Comment by yum-yeom** [comment]
3종(Notebook·WikiDocs·EPUB) × 2인 PASS 충족 확인 — C1 1차(`61a71df`·`30a5cae`), C2 2차(`4863c9e`·`4a3ae77`).
Ch 21 몫으로 열려 있는 결함은 #188(Minor, 동결 전 일괄) 뿐이라 Blocker·Major 0건입니다.
전 챕터 이슈(#210 EPUB `\eqref`, #143·#82 WikiDocs)는 변환기·전역 몫이라 별도 추적합니다. 닫습니다.
